### PR TITLE
[WIP]  9.0 improvement

### PR DESCRIPTION
**Pull request by sebastienbeau**
@simahawk @guewen 
Hi I have done some (lot) refactor in order to have more information.
- having the detail for http request
- having data about time and number of sql request (slow and normal request)
- having data for the cron (state, timing)

I have done the PR in 9 because is the older version and I need to make it working on 8 (the module 9 work perfectly on 8).

I will totally understand if you do not want to merge this change as there broke the current version.
The aim of my PR is more to show you what I am working on and get your feedback.

TODO
-  [ ] metric for queue_job (I think I will add an extra module)

### CODE REVIEW

**Comment by simahawk** [comment]
@sebastienbeau what about a README and an example of output? :)

**Comment by guewen** [comment]
Very interesting! You might want to have a look at https://github.com/camptocamp/odoo-cloud-platform/pull/37 which is close in the goal (with a different tooling though).

Out of curiosity, did you benchmark the overhead, especially on `Cursor.execute` which is called a lot?

File: monitoring_statsd/models/ir_http.py [hunk 1]
```diff
@@ -16,26 +22,37 @@ def _dispatch(self):
...
+        name = '.'.join([
+            path_info.replace('.', '-'),
+            customer,
+            environment,
+            request.params.get('model', 'undefined').replace('.', '_'),
+            action
+            ])
+
+        with statsd.pipeline() as pipe:
+            start = datetime.now()
```
Comment (guewen) [inline monitoring_statsd/models/ir_http.py:50]: I would use `time.time()`

File: monitoring_statsd/sql_tracker.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            if surfixe:
+                prefix = "{}_{}".format(prefix, surfixe)
+            if name:
+                return "{}.{}".format(prefix, name)
+            else:
+                return prefix
+
+        pipe.timing(metric_name(prefix, None, name), self.duration)
+        pipe.incr(metric_name(prefix, 'total', name), self.count)
+        if self.slow_active:
```
Comment (guewen) [inline monitoring_statsd/sql_tracker.py:46]: I'm not sure how `slow_active` is meant to be used

File: monitoring_statsd/sql_tracker.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+
+ori_execute = Cursor.execute
+
+
+def execute(self, query, params=None, log_exceptions=None):
+    start = datetime.now()
+    res = ori_execute(
+        self, query, params=params, log_exceptions=log_exceptions)
+    end = datetime.now()
```
Comment (guewen) [inline monitoring_statsd/sql_tracker.py:67]: I would use `time.time()`, again ;)

File: monitoring_statsd/sql_tracker.py (hunk 2 above)
Comment (guewen) [reply monitoring_statsd/sql_tracker.py:46]: Ok, got it :)

**Pull request closed without merging**